src/main/kotlin/ecommerce/service/OrderService.kt

```diff
@@ -0,0 +1,164 @@
+package ecommerce.service
+
+import ecommerce.dto.OrderResponse
+import ecommerce.dto.PaymentRequest
+import ecommerce.dto.PaymentResponse
+import ecommerce.dto.PlaceOrderRequest
+import ecommerce.exception.OrderProcessingException
+import ecommerce.exception.StripePaymentException
+import ecommerce.model.Member
+import ecommerce.model.Order
+import ecommerce.model.OrderItem
+import ecommerce.model.Payment
+import ecommerce.model.ProductOption
+import ecommerce.repository.CartItemRepository
+import ecommerce.repository.MemberRepository
+import ecommerce.repository.OrderRepository
+import ecommerce.repository.ProductOptionRepository
+import ecommerce.stripe.DeclineCode
+import ecommerce.stripe.StripeClient
+import ecommerce.utils.ResponseMapper.orderToResponse
+import jakarta.persistence.EntityNotFoundException
+import jakarta.transaction.Transactional
+import org.springframework.data.domain.Page
+import org.springframework.data.domain.PageRequest
+import org.springframework.data.domain.Sort
+import org.springframework.data.repository.findByIdOrNull
+import org.springframework.stereotype.Service
+import org.springframework.web.client.RestClient
+import java.time.LocalDateTime
+
+@Service
+class OrderService(
+    private val orderRepository: OrderRepository,
+    private val productOptionRepository: ProductOptionRepository,
+    private val cartItemRepository: CartItemRepository,
+    private val memberRepository: MemberRepository,
+    private val stripeClient: StripeClient,
+) {
+    private val restClient = RestClient.create()
+
+    fun getMemberOrders(
+        memberId: Long,
+        page: Int,
+        size: Int,
+        sortBy: String,
+    ): Page<OrderResponse> {
+        validateMember(memberId)
+        val pageable = PageRequest.of(page, size, Sort.by(sortBy))
```

Comment:
Providing what can be selected with `sortBy` as an enum would make it easier for the client to use and also easier for the server to validate. 🙂